src/cascadia/TerminalApp/Tab.h

```diff
@@ -37,6 +39,7 @@ class Tab
 
 private:
     std::shared_ptr<Pane> _rootPane{ nullptr };
+    winrt::hstring _lastIconPath{};

```

Comment:
i'd like to move towards only using hstring at projection boundaries -- it's less allocations and more manageable to use wstrings. thoughts?

Reply:
I'm tempted to make Tab a winrt type to have an ObservableProperty IconPath, so it'll just update automagically. So this is like, half way there

Reply:
I reckon once we do that we can actually set the _tab object_ to be the "Header" in the TabViewItem.

Reply:
my hugest sadness with `winrt::hstring` is that we _must_ (are contractually obligated to) use the less efficient type every time we use it.

`winrt::param::hstring` automatically handles wstring_view and wstring and stuff -- it does some magic to save allocations and everything. _It is also in a private namespace._

`winrt::hstring` forces an allocation of a real HSTRING ☹️